### PR TITLE
Small code enhancements upon Search feature codebase

### PR DESCRIPTION
**Pull request by KevinBoulongne**
- LocalSettings: Add `DEFAULT_RECENT_SEARCHES` value
- FolderController: Clarify `getFoldersQuery()` code
- ThreadController: Rename `apiThreads` into `remoteThreads`
- SearchViewModel: Rename `hasNoNextPage` into `isLastPage`

### CODE REVIEW

**Review by LunarX — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt [hunk 1]
```diff
@@ -178,7 +178,7 @@ class LocalSettings private constructor(context: Context) {
 
     //region Recent searches
     var recentSearches: List<String>
-        get() = json.decodeFromString(sharedPreferences.getString(RECENT_SEARCHES_KEY, "[]") ?: "[]")
+        get() = json.decodeFromString(sharedPreferences.getString(RECENT_SEARCHES_KEY, DEFAULT_RECENT_SEARCHES)!!)
         set(value) = sharedPreferences.transaction { putString(RECENT_SEARCHES_KEY, json.encodeToString(value)) }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:182]: An improvement would be more like this
```kotlin
var recentSearches: Set<String>
        get() = sharedPreferences.getStringSet(RECENT_SEARCHES_KEY, emptySet())!!
        set(value) = sharedPreferences.transaction { putStringSet(RECENT_SEARCHES_KEY, value) }
```

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt [hunk 2]
```diff
@@ -204,11 +204,12 @@ class LocalSettings private constructor(context: Context) {
 
         private val DEFAULT_THREAD_DENSITY = ThreadDensity.LARGE
         private val DEFAULT_THEME = Theme.SYSTEM
-        val DEFAULT_ACCENT_COLOR = AccentColor.PINK
+        private val DEFAULT_ACCENT_COLOR = AccentColor.PINK
         private val INITIAL_SWIPE_ACTION = SwipeAction.TUTORIAL
         val DEFAULT_SWIPE_ACTION_RIGHT = SwipeAction.READ_UNREAD
         val DEFAULT_SWIPE_ACTION_LEFT = SwipeAction.DELETE
         private val DEFAULT_EXTERNAL_CONTENT = ExternalContent.ASK_ME
+        private const val DEFAULT_RECENT_SEARCHES = "[]"
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:212]: It disappears with the previous remark

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:182]: The order is lost with the `Set`.

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:212]: Related to https://github.com/Infomaniak/android-mail/pull/648#discussion_r1119791175

**Pull request merged**